base/number.jl

```diff
@@ -136,6 +136,52 @@ true
 """
 signbit(x::Real) = x < 0
 
+"""
+    ispositive(x)
+
+Test whether `x > 0`. See also [`isnegative`](@ref).
+
+!!! compat "Julia 1.12"
+    This function requires at least Julia 1.12.
+
+# Examples
+```jldoctest
+julia> ispositive(-4.0)
+false
+
+julia> ispositive(99)
+true
+
+julia> ispositive(0.0)
+false
+```
+"""
+ispositive(x) = x > zero(x) # generic fallback
```

Comment:
:-1:

Why is there an `Any` method? The definition mostly makes sense for `Real`.

Reply:
It is because of #35067

Reply:
In any case it should not belong to `number.jl`. If it has to be added, where should it go?

Reply:
> It is because of https://github.com/JuliaLang/julia/issues/35067

I don't see the connection. That issue doesn't seem to discuss that part of the design. It barely discusses design at all.

Reply:
That issue discussed reasons of having the general interface.

Reply:
I agree with @nsajko. It's always possible to add more generic functionality later, but it's impossible to go backwards.  The most generic `zero` gets is `::Number`.  If we have a function in base, then non-numeric types (like Dates) can of course choose to extend it if appropriate.

Reply:
Oh, I get it, thanks.